api/src/main.rs

```diff
@@ -168,50 +165,71 @@ impl From<jwt_simple::Error> for Error {
     }
 }
 
-#[derive(Serialize, Deserialize, ToSchema, ToResponse)]
-#[aliases(
-    OkLoginResponse = OkResponse<LoginResponse>,
-    OkCreateListResponse = OkResponse<CreateListResponse>,
-    OkGetListsResponse = OkResponse<GetListsResponse>,
-    OkSearchAccountResponse = OkResponse<SearchAccountResponse>,
-    OkReadListResponse = OkResponse<ReadListResponse>,
-    OkAddToListResponse = OkResponse<AddToListResponse>,
-    OkGetHistoryResponse = OkResponse<GetHistoryResponse>,
-    OkUpdateItemResponse = OkResponse<UpdateItemResponse>,
-    OkDeleteItemResponse = OkResponse<DeleteItemResponse>,
-    OkDeleteListResponse = OkResponse<DeleteListResponse>,
-    OkUnshareResponse = OkResponse<UnshareResponse>,
-    OkGetSharesResponse = OkResponse<GetSharesResponse>,
-    OkShareListResponse = OkResponse<ShareListResponse>,
-    OkDeleteShareResponse = OkResponse<DeleteShareResponse>,
-    OkRecoveryInfoResponse = OkResponse<RecoveryInfoResponse>,
-    OkRecoverPasswordResponse = OkResponse<RecoverPasswordResponse>,
-    OkRegisterResponse = OkResponse<RegisterResponse>,
-    OkGetAccountNameResponse = OkResponse<GetAccountNameResponse>,
-    OkSetPublicResponse = OkResponse<SetPublicResponse>,
-    OkRemovePublicResponse = OkResponse<RemovePublicResponse>,
-    OkGetPantryResponse = OkResponse<GetPantryResponse>,
-    OkAddToPantryResponse = OkResponse<AddToPantryResponse>,
-    OkRefillPantryResponse = OkResponse<RefillPantryResponse>,
-    OkEditPantryItemResponse = OkResponse<EditPantryItemResponse>,
-    OkDeletePantryItemResponse = OkResponse<DeletePantryItemResponse>,
-)]
-struct OkResponse<T> {
-    ok: T,
+trait OkResponse {
+    type Wrapper;
+
+    fn ok(ok: Self) -> Rsp<Self>;
 }
 
-impl<T> OkResponse<T> {
-    fn ok(v: T) -> Rsp<T> {
-        Ok(Json(Self { ok: v }))
-    }
+macro_rules! alias {
+    ($($okResp:ident => $ty:ident),* $(,)?) => {
+        pub mod ok_response {
```

Comment:
Why do we need to introduce this module? We could simply dump everything in the main namespace, I don't think it's too bad of a problem, as everything seems to include crate::ok_response::*

Reply:
I didn't particularly like the aspect of `use crate::*` in every modules. Here the `crate::ok_response::*` is pretty self contained to importing only `Ok...Response`

Reply:
Hmm good point